### PR TITLE
deployments: use events to verify the hooks run and replace --follow with rollout status

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/11887

### CODE REVIEW

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -236,18 +237,28 @@ var _ = g.Describe("deploymentconfigs", func() {
 			o.Expect(config.Spec.Test).Should(o.BeTrue())
 
 			g.By("deploying a few more times")
+			// expression to match the name of the rolled deployment
+			deploymentVersionExp := regexp.MustCompile(`"deployment-test-(\d+)" has completed progressing`)
```
Comment (mfojtik) [inline test/extended/deployments/deployments.go]: @kargakis are the version numbers reliable (i+2) (iow. i don't need this regexp)

**Comment by mfojtik** [comment]
[test]

File: test/extended/deployments/deployments.go [hunk 2]
```diff
@@ -237,17 +237,19 @@ var _ = g.Describe("deploymentconfigs", func() {
...
 
-				g.By(fmt.Sprintf("checking the logs for substrings\n%s", out))
-				o.Expect(out).To(o.ContainSubstring(fmt.Sprintf("deployment-test-%d up to 1", i+2)))
-				o.Expect(out).To(o.ContainSubstring("--> pre: Success"))
-				o.Expect(out).To(o.ContainSubstring("test pre hook executed"))
-				o.Expect(out).To(o.ContainSubstring("--> Success"))
+				g.By("checking the events for deployment hooks")
+				out, err = oc.Run("describe").Args("dc/deployment-test").Output()
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(out).To(o.ContainSubstring(fmt.Sprintf("The pre-hook for deployment %s/deployment-test-%d completed successfully", oc.Namespace(), i+2)))
```
Comment (kargakis) [inline test/extended/deployments/deployments.go]: This test is about checking that a test deployment works, don't we have a separate issue for hook logs? I think so. Iow. you should just use `rollout status` here.

Comment (mfojtik) [inline test/extended/deployments/deployments.go]: @kargakis so we don't need to check the logs at all? i'm OK with that.

**Comment by mfojtik** [comment]
original issue was fixed by https://github.com/openshift/origin/pull/12818

**Pull request closed without merging**